optimize/backend/src/main/java/io/camunda/optimize/tomcat/CcsmRequestAdjustmentFilter.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.tomcat;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+public class CcsmRequestAdjustmentFilter implements Filter {
```

Comment:
What is this filter needed for? Why modify the URI that is being used?

Reply:
@buccarel - thanks for the comprehensive write up here 👍 

Can you create issue(s) for the follow-up tasks you mention here? They could be linked to the epic as follow-up tasks if appropriate, but it would be good to ensure they don't get missed/forgotten

Reply:
Done: https://github.com/camunda/camunda/issues/24910